specification/intelligentdataexperiences/resource-manager/Microsoft.AadCustomSecurityAttributesDiagnosticSettings/preview/2023-03-01-preview/intelligentdataexperiences.json

```diff
@@ -0,0 +1,486 @@
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/NameParameter"
+          }
+        ],
+        "responses": {
+          "default": {
+            "description": "Error. Request for AadCustomSecurityAttributesDiagnosticSettings diagnostic setting has failed.",
+            "schema": { "$ref": "#/definitions/ErrorResponse" }
+          },
+          "200": {
+            "description": "Successful request to get more information about diagnostic setting",
+            "schema": { "$ref": "#/definitions/DiagnosticSettingsResource" }
+          }
+        },
+        "x-ms-examples": {
+          "Gets the diagnostic setting": { "$ref": "./examples/getDiagnosticSetting.json" }
+        }
+      },
+      "put": {
+        "tags": ["DiagnosticSettings"],
+        "operationId": "DiagnosticSettings_CreateOrUpdate",
+        "x-ms-examples": {
+          "BatchAccountDelete": { "$ref": "./examples/createOrUpdateDiagnosticSetting.json" }
+        },
+        "description": "Creates or updates diagnostic settings for AadCustomSecurityAttributesDiagnosticSettings.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": { "$ref": "#/definitions/DiagnosticSettingsResource" },
+            "description": "Parameters supplied to the operation."
+          },
+          {
+            "$ref": "#/parameters/NameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successful request to create AadCustomSecurityAttributesDiagnosticSettings diagnostic setting.",
+            "schema": { "$ref": "#/definitions/DiagnosticSettingsResource" }
+          },
+          "default": {
+            "description": "Error. The request to create diagnostic setting has failed.",
+            "schema": { "$ref": "#/definitions/ErrorResponse" }
+          }
+        }
+      },
+      "delete": {
+        "tags": ["DiagnosticSettings"],
+        "operationId": "DiagnosticSettings_Delete",
+        "description": "Deletes existing diagnostic setting for AadCustomSecurityAttributesDiagnosticSettings.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/NameParameter"
+          }
+        ],
+        "responses": {
+          "default": {
+            "description": "Error. The request to delete named diagnostic setting has failed.",
+            "schema": { "$ref": "#/definitions/ErrorResponse" }
+          },
+          "200": { "description": "Successful request to remove a diagnostic setting" },
+          "204": { "description": "Successful request to remove a diagnostic setting" }
```

Comment:
>Successful request to remove a diagnostic setting [](http://example.com/codeflow?start=35&length=49)

204 means the diagnostic setting resource is not found or is already deleted.

Reply:
Did some searching on this, google seems to suggest otherwise. I added "No content" to the response though.